packages/components/src/box-control/input-controls.js

```diff
@@ -66,51 +92,72 @@ export default function BoxInputControls( {
 		handleOnChange( nextValues );
 	};
 
+	const isSideDisabled = ( side ) => sides && sides[ side ] === false;
+	const { first, last, only } = getFirstLastAndOnlySides( sides );
+
 	return (
 		<LayoutContainer className="component-box-control__input-controls-wrapper">
 			<Layout
 				gap={ 0 }
 				align="top"
 				className="component-box-control__input-controls"
 			>
-				<UnitControl
-					{ ...props }
-					isFirst
-					value={ top }
-					onChange={ createHandleOnChange( 'top' ) }
-					onFocus={ createHandleOnFocus( 'top' ) }
-					onHoverOn={ createHandleOnHoverOn( 'top' ) }
-					onHoverOff={ createHandleOnHoverOff( 'top' ) }
-					label={ LABELS.top }
-				/>
-				<UnitControl
-					{ ...props }
-					value={ right }
-					onChange={ createHandleOnChange( 'right' ) }
-					onFocus={ createHandleOnFocus( 'right' ) }
-					onHoverOn={ createHandleOnHoverOn( 'right' ) }
-					onHoverOff={ createHandleOnHoverOff( 'right' ) }
-					label={ LABELS.right }
-				/>
-				<UnitControl
-					{ ...props }
-					value={ bottom }
-					onChange={ createHandleOnChange( 'bottom' ) }
-					onFocus={ createHandleOnFocus( 'bottom' ) }
-					onHoverOn={ createHandleOnHoverOn( 'bottom' ) }
-					onHoverOff={ createHandleOnHoverOff( 'bottom' ) }
-					label={ LABELS.bottom }
-				/>
-				<UnitControl
-					{ ...props }
-					isLast
-					value={ left }
-					onChange={ createHandleOnChange( 'left' ) }
-					onFocus={ createHandleOnFocus( 'left' ) }
-					onHoverOn={ createHandleOnHoverOn( 'left' ) }
-					onHoverOff={ createHandleOnHoverOff( 'left' ) }
-					label={ LABELS.left }
-				/>
+				{ ! isSideDisabled( 'top' ) && (
```

Comment:
It feels like we could just loop over the "sides" if that prop is transformed to be an array. wdyt?

Reply:
Has the potential to clean this up. I'm happy to iterate on this. Thanks!

Reply:
✅ Done.